src/BasicVault.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "../lib/yield-utils-v2/contracts/token/IERC20.sol";
+import "../lib/yield-utils-v2/contracts/token/IERC2612.sol";
+
+
+/// @title Basic Vault
+/// @author hashedMae
+/// @notice A simple vault that tracks the amount of tokens a user has deposited and allows them to withdraw at any time
+/// @dev 
+
+contract BasicVault {
+
+IERC20 iToken;
+
+mapping(address => uint256) public userBalance;
```

Comment:
I would probably name this mapping `balances`. It's customary to use plurals in mappings, and I prefer to use less words in variable names if that doesn't lead to confusion.